bin/spack-tmpconfig

```diff
@@ -72,6 +72,7 @@ config:
     root: $TMP_DIR/install
   misc_cache: $$user_cache_path/cache
   source_cache: $$user_cache_path/source
+  environments_root: $TMP_DIR/envs
```

Comment:
I'm having trouble tracking why this change is necessary.

Reply:
This was a suggestion from @trws to sandbox parallel execution. I don't fully understand it either.  @trws perhaps you could comment?

Reply:
That script is used to separate the unit tests when run in parallel.  It ensures that each test runner gets its own separate spack environment so they don't tromp each other.  If the environments_root defaults to being under user_cache_path or something then it's unnecessary, but if it's in the spack_root by default or something like that then the script needs to put it somewhere else.

Reply:
Yeah @psakievich and I came to the conclusion that this probably isn't *technically* necessary since all of the environment tests use a mock environment root anyway, but that we should keep it to ensure future tests are separated for parallel execution.

Reply:
Sounds right to me.  The script is meant to be a safety net and a workaround for all the tests that haven't been converted yet. It's also oddly useful as a way to do one-off testing without messing up a current environment just on the command line, so having it there helps with that kind of use.